R/PipeOpLearnerCVPlus.R

```diff
@@ -0,0 +1,204 @@
+#'   [`Learner`][mlr3::Learner] that is being wrapped. Read-only.
+#' * `learner_model` :: [`Learner`][mlr3::Learner]\cr
+#'   [`Learner`][mlr3::Learner] that is being wrapped. This learner contains the model if the `PipeOp` is trained. Read-only.
+#' * `predict_type`\cr
+#'   Predict type of the [`PipeOpLearnerCVPlus`], which is always `"response"  "quantiles"`.
+#'   This can be different to the predict type of the [`Learner`][mlr3::Learner] that is being wrapped.
+#'
+#' @section Methods:
+#' Methods inherited from [`PipeOp`].
+#'
+#' @references
+#' `r format_bib("barber_2021")`
+#'
+#' @family PipeOps
+#' @family Meta PipeOps
+#' @template seealso_pipeopslist
+#' @include PipeOp.R
+#' @export
+#' @examples
+#' \dontshow{ if (requireNamespace("rpart")) \{ }
+#' library("mlr3")
+#'
+#' task = tsk("mtcars")
+#' learner = lrn("regr.ranger")
+#' lrncvplus_po = mlr_pipeops$get("learner_cv_plus", learner)
+#'
+#' lrncvplus_po$train(list(task))
+#' lrncvplus_po$predict(list(task))
+#' \dontshow{ \} }
+PipeOpLearnerCVPlus = R6Class("PipeOpLearnerCVPlus",
+  inherit = PipeOp,
+  public = list(
+    initialize = function(learner, id = NULL, param_vals = list()) {
+      private$.learner = as_learner(learner, clone = TRUE)
+      id = id %??% private$.learner$id
+      type = private$.learner$task_type
+
+      if (!("regr" %in% type)) {
+        stop("PipeOpLearnerCVPlus only supports regression.")
+      }
+
+      task_type = mlr_reflections$task_types[type, mult = "first"]$task
+      out_type = mlr_reflections$task_types[type, mult = "first"]$task
+
+      # paradox requirements 1.0
+      private$.cvplus_param_set = ps(
+        folds = p_int(lower = 2L, upper = Inf, tags = c("train", "required")),
+        alpha = p_dbl(lower = 0L, upper = 1L, tags = c("predict", "required"))
+      )
+
+      private$.cvplus_param_set$values = list(folds = 3, alpha = 0.05) # default
+
+      super$initialize(id, param_set = alist(cvplus = private$.cvplus_param_set, private$.learner$param_set),
+                        param_vals = param_vals,
+                        input = data.table(name = "input", train = task_type, predict = task_type),
+                        output = data.table(name = "output", train = "NULL", predict = out_type),
+                        packages = learner$packages,
+                        tags = c("learner", "ensemble")
+      )
+    }
+  ),
+  active = list(
+    learner = function(val) {
+      if (!missing(val)) {
+        if (!identical(val, private$.learner)) {
+          stop("$learner is read-only.")
+        }
+      }
+      private$.learner
+    },
+    learner_model = function(val) {
+      if (!missing(val)) {
+        if (!identical(val, private$.learner)) {
+          stop("$learner_model is read-only.")
+        }
+      }
+      if (is.null(self$state) || is_noop(self$state)) {
+        private$.learner
+      } else {
+        multiplicity_recurse(self$state, clone_with_state, learner = private$.learner)
```

Comment:
this does not work, since the PipeOp's `$state` is not the same as the Learner's `$state`; this is different in the PipeOpLearnerCV. But tbh I don't know what kind of result one would want here, maybe best to discuss this.

One possibility of what could work would be
```r
multiplicity_recurse(self$state, learner = function(state) {
  map(state$cv_model_states, clone_with_state, learner = private$.learner)
})
```